Add new user experience cross org documentation (#2790)
sourcegraph/handbook: content/departments/product-engineering/process/new-user-experience.md

No one team or organization can effectively own this entire experience, so careful collaboration will be required to avoid duplication of efforts and help our users find success with Sourcegraph.

## Cloud responsibilities

### Growth 
The Growth team will own the initial journeys of both new unauthenticated and authenticated users take to discover key functionality and use cases. Typically these journeys will typically provide an introduction to major product differentiators and features that factor heavily in user activation and retention. This includes the unauthenticated homepage, product tour and/or quick start flows, sign-up flows, and the post sign-up flow. 

From time to time, the Growth team will need to alter these user journeys to accommodate advances in product functionality. 

The scope of Growth's efforts should not extend the specifics of how to use individual product features. 

### Cloud SaaS
The Cloud team will own user journeys related to the onboarding of teams. This includes flows related to team creation, user invites, and administration. 

## Code Graph responsibilities

> Search, Batch Changes, Code Insights, and Code Intelligence will own user education for their specific product areas. They will  collaborate with other code graph teams to promote feature discovery in places like search results, repository pages, and other areas of the site where opportunities exist for integrations.

Author: ```suggestion
Search, Batch Changes, Code Insights, and Code Intelligence will own user education for their specific product areas. They will  collaborate with other code graph teams to promote feature discovery in places like search results, repository pages, and other areas of the site where integration opportunities exist.
```

# Collaborating on the new user experience

The new user experience (NUE) crosses the bounds of several teams' ownership areas. For example, a visitor to the product may view the home page, search several times, learn about Code Insights and then sign up for the product. Throughout this journey, they should be informed as to the major functionality and use cases of Sourcegraph as well as how to use each feature.

> No one team or organization can effectively own this entire experience, so careful collaboration will be required to avoid duplication of efforts and help our users find success with Sourcegraph.

Reviewer: ```suggestion
No one team or organization can effectively own this entire experience, so careful collaboration will be required to avoid duplication of efforts and help our users find success with Sourcegraph.

Major changes or more risky changes, such as adding lots of CTAs in key areas, should ideally be feature-flagged and A/B tested before being rolled out to the full fleet of customers. That way we maximize learning speed (no long-ish review process), but still do so in a way that is safe for the overall user experience.
```

Reviewer: @malomarrec added your suggestion as a suggestion - looks good?

Author: I really don't think we should specify how any one team works in this overview, it's only about how we work together.

Reviewer: @rrhyne I'm missing why this would be specific to only one team?

Author: Let me clarify: I don't think we should specify how the initiatives are moved forward. A/B testing is a great practice, but some projects are risky but not A/B testable, such as when there is no baseline, there is not enough traffic for the test to be significant, or the timeline makes it impossible.

Reviewer: That makes sense. Maybe there's another way to convey the point here, which is that we need to make sure we think about the design/user experience, and don't create disjointed or annoying new user experiences given the shared ownership.

Author: I support that but need to go ahead and merge this before I head out on vacation. Happy to create a PR to add this when I get back!

## Cloud responsibilities

> ### Growth

Reviewer: I think this team is called Growth & Integrations

```suggestion
### Growth & Integrations
```

> The Growth team will own the initial journeys of both new unauthenticated and authenticated users take to discover key functionality and use cases. Typically these journeys will typically provide an introduction to major product differentiators and features that factor heavily in user activation and retention. This includes the unauthenticated homepage, product tour and/or quick start flows, sign-up flows, and the post sign-up flow.

Reviewer: ```suggestion
The Growth team will own the initial journeys of both new unauthenticated and authenticated users take to discover key functionality and use cases. Typically these journeys will provide an introduction to major product differentiators and features that factor heavily in user activation and retention. This includes the unauthenticated homepage, product tour and/or quick start flows, sign-up flows, and the post sign-up flow.
```

From time to time, the Growth team will need to alter these user journeys to accommodate advances in product functionality.

> The scope of Growth's efforts should not extend the specifics of how to use individual product features.

Reviewer: ```suggestion
The scope of Growth's efforts should not extend the specifics of how to use individual product features, that would be up to the team that owns that feature.
```

> The scope of Growth's efforts should not extend the specifics of how to use individual product features, that would be up to the team that owns that feature.

Reviewer: ```suggestion
The scope of Growth's efforts should not extend the specifics of how to use individual product features, that would be up to the team that owns that feature. The Growth team is welcome of course to provide feedback to the teams where feature journeys are incomplete.
```

> The scope of Growth's efforts should not extend the specifics of how to use individual product features, that would be up to the team that owns that feature. The Growth team is welcome of course to provide feedback to the teams where feature journeys are incomplete.

Reviewer: ```suggestion
The scope of Growth's efforts should not extend the specifics of how to use individual product features, that would be up to the team that owns that feature. The Growth team is welcome of course to provide feedback to the teams where feature journeys are incomplete, but changes of this kind should not be made without alignment with the owning team.
```

I believe this happened in the past so perhaps worth calling out?

Author: Is feedback a change?  Perhaps this should read:

```suggestion
The scope of Growth's efforts should not extend the specifics of how to use individual product features, that would be up to the team that owns that feature. 

At times, the broad nature of the Growth team's mandate will require that it work within other teams' domains. When this need arises, the Growth team will work with the impacted team by contacting the PM to request reviews of artifacts such as product documents, RFCs, and designs. Additionally, engineers from the affected team should be added as reviewers on any resulting PRs.
```

### Cloud SaaS

The Cloud team will own user journeys related to the onboarding of teams. This includes flows related to team creation, user invites, and administration.

## Code Graph responsibilities

> Search, Batch Changes, Code Insights, and Code Intelligence will own user education for their specific product areas. They will collaborate with other code graph teams to promote feature discovery in places like search results, repository pages, and other areas of the site where integration opportunities exist.

Reviewer: ```suggestion
Search Core, Search Product, Batch Changes, Code Insights, and Code Intelligence will own user education for their specific product areas. They will collaborate with other code graph teams to promote feature discovery in places like search results, repository pages, and other areas of the site where integration opportunities exist.
```

> Search Core, Search Product, Batch Changes, Code Insights, and Code Intelligence will own user education for their specific product areas. They will collaborate with other code graph teams to promote feature discovery in places like search results, repository pages, and other areas of the site where integration opportunities exist.

Reviewer: ```suggestion
Search Core, Search Product, Batch Changes, Code Insights, and Code Intelligence will own user education for their specific product areas. They will collaborate within Code Graph to promote feature discovery in places like search results, repository pages, and other areas of the application where integration opportunities exist.
```
